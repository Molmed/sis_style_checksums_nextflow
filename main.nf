process CREATE_SIS_STYLE_CHECKSUMS {
    publishDir "${params.input_folder}/MD5/", mode: 'copy', overwrite: true

    input:
    path input_folder
    path include_file
    val ignore_cache

    output:
    path "checksums.md5"

    script:
    """
    create_sis_style_checksums.sh $input_folder $include_file $ignore_cache
    """
}

workflow {
    CREATE_SIS_STYLE_CHECKSUMS(
        params.input_folder,
        params.include_file,
        params.ignore_cache,
    )
}
