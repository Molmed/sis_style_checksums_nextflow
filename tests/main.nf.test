nextflow_process {
    name "Test Process CREATE_SIS_STYLE_CHECKSUMS"
    script "main.nf"
    process "CREATE_SIS_STYLE_CHECKSUMS"

    test("Generate checksums from scratch") {
        when {
            params {
                input_folder = "${projectDir}/tests/test_resources/runfolder_nocache"
            }
            process {
                """
                input[0] = "${projectDir}/tests/test_resources/runfolder_nocache"
                input[1] = "${projectDir}/tests/test_resources/include.rsync"
                input[2] = true
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("Generate checksums with cache") {
        when {
            params {
                input_folder = "${projectDir}/tests/test_resources/runfolder_cache"
            }
            process {
                """
                input[0] = "${projectDir}/tests/test_resources/runfolder_cache"
                input[1] = "${projectDir}/tests/test_resources/include.rsync"
                input[2] = false
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
