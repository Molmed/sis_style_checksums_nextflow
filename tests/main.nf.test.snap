{
    "Generate checksums from scratch": {
        "content": [
            {
                "0": [
                    "checksums.md5:md5,75a433b1e14782f06d44c93142a2d3fb"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-09T13:40:49.058439"
    },
    "Generate checksums with cache": {
        "content": [
            {
                "0": [
                    "checksums.md5:md5,d6a4e7c087ac3895562df860e16a9c10"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-09T13:40:53.342759"
    }
}